File: tests/converter_test.py
import unittest
from pandas import read_csv

# import sys
# sys.path.append('../src')
# from src import converter
# Converter = converter.Converter
# Alphabet = converter.Alphabet
from src.converter import Converter, Alphabet

class TestConvert(unittest.TestCase):
    def test_convert(self):
        b62_to_b42 = Converter(Alphabet.Base62.value, Alphabet.Base42.value)
        b42_to_b62 = Converter(Alphabet.Base42.value, Alphabet.Base62.value)

        for _, row in read_csv('./tests/fixtures/data.csv').iterrows():
            converted = b62_to_b42.convert(row['base62'])
            self.assertEqual(converted, row['base42'])
            round_tripped = b42_to_b62.convert(converted)
            self.assertEqual(round_tripped, row['base62'])

if __name__ == '__main__':
    unittest.main()

File: tests/converter_test.cs
using System;
using static Converter;
using System.IO;

class Test {
	static void Main() {
		var b62ToB42 = new Converter(Alphabet.Base62, Alphabet.Base42);
		var b42ToB62 = new Converter(Alphabet.Base42, Alphabet.Base62);

		var reader = new StreamReader("./tests/fixtures/data.csv");
		var isHeader = true;
		while (true) {
			var line = reader.ReadLine();

			if (isHeader) {
				isHeader = false;
				continue;
			}

			if (line == null) break;
			if (line == "") continue;

			var cols = line.Split(",");
			var id = cols[0].Trim();
			var b42 = cols[1].Trim();

			var converted = b62ToB42.Convert(id);

			if (converted != b42) {
				throw new Exception($"{converted} != {b42}");
			}

			var roundTripped = b42ToB62.Convert(converted);

			if (roundTripped != id) {
				throw new Exception($"{roundTripped} != {id}");
			}
		}
		reader.Close();

		Console.WriteLine($"All tests passed ✅");
	}
}
